crates/ruma-events/src/media_preview_config.rs

```diff
@@ -0,0 +1,132 @@
+//! Types for the [`m.media_preview_config`] event.
+//!
+//! [`m.media_preview_config`]: https://github.com/matrix-org/matrix-spec-proposals/pull/4278
+
+use ruma_macros::StringEnum;
+use serde::{Deserialize, Serialize};
+
+use crate::{macros::EventContent, PrivOwnedStr};
+
+/// The content of an `io.element.msc4278.media_preview_config` event,
+/// the unstable version of `m.media_preview_config` in global account data.
+#[derive(Clone, Debug, Deserialize, Serialize, EventContent)]
+#[cfg_attr(not(ruma_unstable_exhaustive_types), non_exhaustive)]
+#[ruma_event(type = "io.element.msc4278.media_preview_config", kind = GlobalAccountData)]
```

Comment:
I see that this type should be available both as global account data and room account data…

If you don't care about the room account data right now it's fine, but we might want to improve our macro to be able to declare this as both kinds, because I don't think that it's possible currently.

Reply:
for now I'll focus on the global account data implementation which is the highest priority one